katharsis-core/src/main/java/io/katharsis/queryspec/InMemoryEvaluator.java

```diff
@@ -97,11 +97,15 @@ private static boolean matchesPrimitiveOperator(Object object, FilterSpec filter
 		Object value = PropertyUtils.getProperty(object, filterSpec.getAttributePath());
 		FilterOperator operator = filterSpec.getOperator();
 		Object filterValue = filterSpec.getValue();
-		if (value instanceof Collection) {
-			return matchesAny((Collection<?>) value, operator, filterValue);
-		}
-		else {
-			return operator.matches(value, filterValue);
+		if (filterValue != null ||  (filterValue == null && (operator.equals(FilterOperator.EQ) || operator.equals(FilterOperator.NEQ)))) {
```

Comment:
You should look [this line](https://github.com/apetrucci/katharsis-framework/blob/master/katharsis-core/pom.xml#L8) where Katharsis-core pom has the relativePath to "../katharsis-parent".  

The katharsis-parent pom sets the javaVersion to 1.7 at [this line](https://github.com/apetrucci/katharsis-framework/blob/master/katharsis-parent/pom.xml#L35).

Therefore I get this error if I use the lambda expression:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /afs/cern.ch/user/a/apetro/mygit/katharsis-framework/katharsis-core/src/main/java/io/katharsis/queryspec/InMemoryEvaluator.java:[103,104] lambda expressions are not supported in -source 1.7
  (use -source 8 or higher to enable lambda expressions)
[INFO] 1 error

```